src/dotnet/CarbonAware.Aggregators/src/CarbonAware/CarbonAwareAggregator.cs

```diff
@@ -2,28 +2,73 @@
 using CarbonAware.Interfaces;
 using Microsoft.Extensions.Logging;
 using System.Collections;
+using System.Diagnostics;
 
 namespace CarbonAware.Aggregators.CarbonAware;
 
 public class CarbonAwareAggregator : ICarbonAwareAggregator
 {
+    private static readonly ActivitySource Activity = new ActivitySource(nameof(CarbonAwareAggregator));
     private readonly ILogger<CarbonAwareAggregator> _logger;
-
     private readonly ICarbonIntensityDataSource _dataSource;
 
+    /// <summary>
+    /// Creates a new instance of the <see cref="CarbonAwareAggregator"/> class.
+    /// </summary>
+    /// <param name="logger">The logger for the aggregator</param>
+    /// <param name="dataSource">An <see cref="ICarbonIntensityDataSource"> data source.</param>
     public CarbonAwareAggregator(ILogger<CarbonAwareAggregator> logger, ICarbonIntensityDataSource dataSource)
     {
         this._logger = logger ?? throw new ArgumentNullException(nameof(logger));
         this._dataSource = dataSource;
     }
 
+    /// <inheritdoc />
     public async Task<IEnumerable<EmissionsData>> GetEmissionsDataAsync(IDictionary props)
     {
-        DateTimeOffset end = GetOffsetOrDefault(props, CarbonAwareConstants.End, DateTimeOffset.Now);
-        DateTimeOffset start = GetOffsetOrDefault(props, CarbonAwareConstants.Start, end.AddDays(-7));
-        _logger.LogInformation("Aggregator getting carbon intensity from data source");
-        return await this._dataSource.GetCarbonIntensityAsync(GetLocationOrThrow(props), start, end);
+        using (var activity = Activity.StartActivity())
+        {
+            DateTimeOffset end = GetOffsetOrDefault(props, CarbonAwareConstants.End, DateTimeOffset.Now);
+            DateTimeOffset start = GetOffsetOrDefault(props, CarbonAwareConstants.Start, end.AddDays(-7));
+            _logger.LogInformation("Aggregator getting carbon intensity from data source");
+            return await this._dataSource.GetCarbonIntensityAsync(GetLocationOrThrow(props), start, end);
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task<IEnumerable<EmissionsForecast>> GetCurrentForecastDataAsync(IDictionary props)
+    {
+        using (var activity = Activity.StartActivity())
+        {
+            DateTimeOffset start = GetOffsetOrDefault(props, CarbonAwareConstants.Start, DateTimeOffset.Now);
+            DateTimeOffset end = GetOffsetOrDefault(props, CarbonAwareConstants.End,  start.AddDays(1));
+            _logger.LogInformation("Aggregator getting carbon intensity forecast from data source");
```

Comment:
Probably a future work item, but we need to find some way of maybe standardizing these messages in the future.

Reply:
Agreed!  Out of scope for this PR but worthy of our time in the Logging and Telemetry story.